agent/jvm/src/main/kotlin/io/mockk/proxy/jvm/advice/ValueClassSupport.kt

```diff
@@ -0,0 +1,69 @@
+package io.mockk.proxy.jvm.advice
+
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.full.primaryConstructor
+import kotlin.reflect.jvm.isAccessible
+
+private val valueClassFieldCache = mutableMapOf<KClass<out Any>, KProperty1<out Any, *>>()
+
+/**
+ * @return boxed value of value class if this is value class, else just itself
+ */
+fun <T : Any> T.boxedValue(): Any? {
+    if (!this::class.isValueClass()) return this
+
+    // get backing field
+    val backingField = this::class.valueField()
+
+    // get boxed value
+    @Suppress("UNCHECKED_CAST")
+    return (backingField as KProperty1<T, *>).get(this)
+}
+
+fun <T : Any> T.boxedClass(): KClass<*> {
```

Comment:
Could you add a brief kdoc on this since it's a public api?

Reply:
Should it be part of public API? Couldn't it be marked as `internal`? It doesn't feel like something a user of mockk expects to find. Just my 2 cents :)

Reply:
@Kantis I'd like to make it internal, however it is used within `mockk-agent-jvm` and `mockk-jvm` module. Any ideas how to solve this?

Reply:
Hmm, actually I think it'll be fine as-is.

Since the declaration lives in `mockk-agent-jvm` and it's included in `mockk-jvm` using `implementation project(":mockk-agent-jvm")`, the public declarations from `mockk-agent-jvm` won't be visible to users of `mockk`.

Only options to mark it as `internal` is to copy the code to both modules or merge the modules.